File: scratch/fix-amp.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, '..', 'src', 'lib', 'blog-templates.ts');
let content = fs.readFileSync(filePath, 'utf8');

// Replace the incorrect escape mapping
// The file has: '&': '&'  (ampersand mapped to itself)
// We need:     '&': '&'  (ampersand mapped to HTML entity)
const oldStr = "'&': '&'";
const newStr = "'&': '" + '&' + 'amp;' + "'";
content = content.replace(oldStr, newStr);

fs.writeFileSync(filePath, content);
console.log('Fixed!');

File: scratch/fix-amp.py
import sys
with open('src/lib/blog-templates.ts', 'r', encoding='utf-8') as f:
    c = f.read()
c = c.replace("'&': '&'", "'&': '&'")
with open('src/lib/blog-templates.ts', 'w', encoding='utf-8') as f:
    f.write(c)
print('Fixed!')
